query/src/main/scala/filodb/query/exec/PromQlExec.scala

```diff
@@ -66,7 +64,7 @@ case class PromQlExec(id: String,
         override def key: RangeVectorKey = CustomRangeVectorKey(r.metric.map (m => m._1.utf8 -> m._2.utf8))
 
         override def rows: Iterator[RowReader] = {
-          samples.iterator.map { v =>
+          samples.iterator.collect { case v: Sampl =>
```

Comment:
More thought needed here.

PromQLExec should invoke with histogramMap = true and then detect histogram types in output to facilitate cross-cluster query composition.

Reply:
I'm adding a TODO with notes.  Not easy as the current output has no provision for schema, and unless we add that creation of RangeVectors would be difficult.